includes/validation/class-amp-validation-manager.php

```diff
@@ -220,6 +222,49 @@ function( $query_vars ) {
 		}
 	}
 
+	/**
+	 * Determine if a post supports AMP validation.
+	 *
+	 * @since 1.2
+	 *
+	 * @param WP_Post|int $post Post.
+	 * @return bool Whether post supports AMP validation.
+	 */
+	public static function post_supports_validation( $post ) {
+		$post = get_post( $post );
+		if ( ! $post ) {
+			return false;
+		}
+
+		$supports_validation = (
+			// Skip if the post type is not viewable on the frontend, since we need a permalink to validate.
+			is_post_type_viewable( $post->post_type )
+			&&
+			! wp_is_post_autosave( $post )
+			&&
+			! wp_is_post_revision( $post )
+			&&
+			'auto-draft' !== $post->post_status
+			&&
+			'trash' !== $post->post_status
+		);
+		if ( ! $supports_validation ) {
+			return false;
+		}
+
+		// Story post type always supports validation.
+		if ( AMP_Story_Post_Type::POST_TYPE_SLUG === $post->post_type ) {
```

Comment:
Don't the conditions determining `$supports_validation` apply for Stories as well? Wondering which one doesn't and therefore if this point is reached.

Reply:
Yes they do, so they're checked above:

https://github.com/ampproject/amp-wp/blob/6e12cf1cf80d9f5df9181ae37854c8d3143c9318/includes/validation/class-amp-validation-manager.php#L239-L253

Maybe I don't understand?